packages/core/src/app/checkout/AnalyticsEvents.ts

```diff
@@ -21,34 +19,79 @@ export enum GuestCheckoutEvents {
   Exit = "Exit"
 }
 
+const defaultTimeout = 1000;
+const defaultPollingInterval = 10;
+// borrowed polling mechanism 
+async function pollUntilAccepted<T>(
+  fn: () => T,
+  accept: (t: T) => boolean,
```

Comment:
thoughts on combining this such that it polls for `timeoutMillis` milliseconds or if fn returns a non-null value? There's only a single use case right now and starting out simpler is less risky